crates/network-scanner/src/ping.rs

```diff
@@ -0,0 +1,49 @@
+use std::net::Ipv4Addr;
+
+use network_scanner_net::tokio_raw_socket::TokioRawSocketStream;
+use network_scanner_proto::icmp_v4;
+
+use tokio::io::{AsyncReadExt, AsyncWriteExt};
+use tracing::trace;
+
+pub async fn ping(ip: Ipv4Addr) -> anyhow::Result<()> {
+    let mut socket = TokioRawSocketStream::connect(ip)
+        .await
+        .map_err(|e| anyhow::anyhow!(e))?;
+
+    let time = std::time::SystemTime::now()
+        .duration_since(std::time::UNIX_EPOCH)
+        .map_err(|e| anyhow::anyhow!(e))?
```

Comment:
suggestion: Re-introduce the context strings, but using the `anyhow::Context` trait. Same at other places.
```suggestion
        .context("cannot access system time")?
```